flang/lib/Frontend/FrontendActions.cpp

```diff
@@ -175,6 +175,45 @@ getExplicitAndImplicitAMDGPUTargetFeatures(CompilerInstance &ci,
   return llvm::join(featuresVec, ",");
 }
 
+// Get feature string which represents combined explicit target features
+// for NVPTX and the target features specified by the user/
+// TODO: Have a more robust target conf like `clang/lib/Basic/Targets/NVPTX.cpp`
```

Comment:
If this code can be refactored and placed in `llvm/lib/Frontend/Driver` then both clang and flang can share it.

Reply:
Sure, it should be doable, `AMDGPU` also needs a similar refactor, however, I'm thinking (and waiting) in adding them to `llvm/offload` as MLIR would also benefit from these refactors.